src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/ClusterLocalRegistrar.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    /// <summary>
+    /// The registrar for the Cluster-Local Registration Strategy.
+    /// </summary>
+    internal class ClusterLocalRegistrar : IGrainRegistrar
```

Comment:
all methods of this class do not need to be async (there is no async code in any of them, plus the name is ...Local...). And it actually confused me in the places where it was called, as the convention is that async does some remote activity, while this is pure local registrar.

Looked more now: `IGrainRegistrar` is for multi cluster support, I get it.
How about we keep the IGrainRegistrar sync for now, since in this PR you only have one impl. of this interface, and in the next step, when we get to needing it to be async, we will reconsider. It is just hard for me to justify making it async without looking how it will be used and implemented for multi cluster. It will also make it easier to see the deltas needed for adding multi cluster if this is also part of multi cluster PR.